### PR TITLE
Hivesource copyentity

### PR DESCRIPTION
**Pull request by ibuenros**
This PR refactors distcp to use `CopyEntity` instead of `CopyableFile`. `CopyableFile` is a subclass of `CopyEntity`, and the only subclass that gets processed by the Gobblin workflow. Other `CopyEntities` can be used to give information to the publisher.

### CODE REVIEW

**Comment by ibuenros** [comment]
@ydai1124 Can you review?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java [hunk 1]
```diff
@@ -53,18 +46,16 @@
...
 @Getter
 @Setter
-@AllArgsConstructor(access = AccessLevel.PROTECTED)
-@NoArgsConstructor(access = AccessLevel.PRIVATE)
-@EqualsAndHashCode
-@Builder(builderClassName = "Builder", builderMethodName = "_hiddenBuilder")
-public class CopyableFile implements File, HasGuid {
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@EqualsAndHashCode(callSuper = true)
+public class CopyableFile extends CopyEntity implements File, HasGuid {
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java]: `CopyEntity` already implements `HasGuid`

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/converter/DistcpConverter.java [hunk 2]
```diff
@@ -25,6 +24,7 @@
 import gobblin.converter.DataConversionException;
 import gobblin.converter.SchemaConversionException;
 import gobblin.converter.SingleRecordIterable;
+import gobblin.data.management.copy.CopyEntity;
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/converter/DistcpConverter.java]: Is this used?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyEventSubmitterHelper.java [hunk 3]
```diff
@@ -49,13 +50,13 @@ static void submitSuccessfulDatasetPublish(EventSubmitter eventSubmitter, Copyab
...
   static void submitFailedDatasetPublish(EventSubmitter eventSubmitter,
-      CopyableFile.DatasetAndPartition datasetAndPartition) {
+      CopyEntity.DatasetAndPartition datasetAndPartition) {
     eventSubmitter.submit(DATASET_PUBLISHED_FAILED_EVENT_NAME, ImmutableMap.of(DATASET_ROOT_METADATA_NAME,
         datasetAndPartition.getDataset().getDatasetURN()));
   }
 
   /**
-   * Submit an sla event when a {@link CopyableFile} is published. The <code>workUnitState</code> passed should have the
+   * Submit an sla event when a {@link gobblin.data.management.copy.CopyEntity} is published. The <code>workUnitState</code> passed should have the
```
Comment (ydai1124) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyEventSubmitterHelper.java]: Should this remain as `CopyableFile`?

**Comment by ydai1124** [comment]
Minor comments. Overall LGTM.

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java (hunk 1 above)
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java]: Fixed

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/converter/DistcpConverter.java (hunk 2 above)
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/converter/DistcpConverter.java]: Fixed

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyEventSubmitterHelper.java (hunk 3 above)
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyEventSubmitterHelper.java]: Fixed

**Pull request merged**